File: 2023/3-2-230520_0524/9012.cpp
#include <cstdio>

int N;
char s[100];

int check()
{
	int bias = 0;
	for (int i = 0; s[i]; i++) {
		if (s[i] == '(') {
			bias += 1;
		}
		else {
			if (bias == 0)
				return 0;
			bias -= 1;
		}
	}
	return bias == 0;
}

int main()
{
	scanf("%d", &N);
	while (N--) {
		scanf("%s", s);
		printf("%s\n", check() ? "YES" : "NO");
	}
}

File: boj/2023/str-230417_0512/9012.py
N = int(input())

def check(s):
    stack = []
    for c in s:
        if c == "(":
            stack.append(c)
        elif c == ")" and not (len(stack) > 0 and stack.pop() == "("):
            return False
    return len(stack) == 0   

for _ in range(N):
    s = input()
    print("YES" if check(s) else "NO")